CHROMIUM: esdfs: Add splice_read and splice_write file ops

Due to commit 36e2c7421f02 ("fs: don't allow splice read/write without
explicit ops"), using sendfile() or splice() on an esdfs file returns
EINVAL. This is causing a CTS failure on 5.10, since the test tries to
copy a ringtone file from the /data/media filesystem using sendfile().

This change adds splice_read and splice_write file ops, setting them to
use the default helpers.

BUG=b:179621031
TEST=android.appsecurity.cts.ExternalStorageHostTest
	#testExternalStorageReadDefaultUris now passes on 5.10 kernel

Fixes: 36e2c7421f02 ("fs: don't allow splice read/write without explicit ops")
Change-Id: Ic025681edd8119cfa2bdc9402bf51e2ee70a374e
Signed-off-by: Reka Norman <rekanorman@google.com>
Reviewed-on: https://chromium-review.googlesource.com/c/chromiumos/third_party/kernel/+/2796762
Reviewed-by: Gwendal Grignou <gwendal@chromium.org>
Commit-Queue: Gwendal Grignou <gwendal@chromium.org>
Tested-by: Gwendal Grignou <gwendal@chromium.org>
diff --git a/fs/esdfs/file.c b/fs/esdfs/file.c
index c4d1909..27c0702 100644
--- a/fs/esdfs/file.c
+++ b/fs/esdfs/file.c
@@ -450,6 +450,8 @@
 	.fasync		= esdfs_fasync,
 	.read_iter	= esdfs_read_iter,
 	.write_iter	= esdfs_write_iter,
+	.splice_read    = generic_file_splice_read,
+	.splice_write   = iter_file_splice_write,
 };
 
 /* trimmed directory options */
